PSPs/drafts/psp-32.md

```diff
@@ -0,0 +1,837 @@
+	pub major: u64,
+	/// The **minor version** is incremented when a new functionality in a backwards compatible 
+	/// manner is added.
+	pub minor: u64,
+	/// The **patch version** is incremented when a backwards compatible bug is fixed.
+	pub patch: u64,
+}`
+```
+
+#### Registered Extensions
+
+The following table contains the extensions already registered by this standard.
+
+| Ext. ID | Version |  Extension |
+| ------: | :-----: | :--------- |
+| 0x0001  | 1.0.0   | [Batch Extension](extensions/batch.md) |
+| 0x0002  | 1.0.0   | [Chunks Extension](extensions/chunks.md) |
+| 0x0003  | 1.0.0   | [Enumerable Extension](extensions/enumerable.md) |
+| 0x0004  | 0.1.0   | **TBD** Teleportation |
+| 0x0005  | 1.0.0   | [Operator](extensions/operator.md) |
+| 0x0006  |         | **TBD** HTLC support |
+
+
+## Policies
+
+Polices define what is the behaviour of the asset for specific operations.
+
+### Minting Policy
+
+Minting policies determines who can mint new tokens:
+
+```Rust
+enum MintingPolicy {
+	/// Minting is not allowed.
+	NotAllowed,
+
+	/// Only this issuer can mint tokens.
+	Issuer(AccountId),
+
+	/// Only this issuer can mint tokens but just once.
+	/// That ensures that no more tokens will be created after the first minting.
+	OneTimeIssuer(AccountId),
+
+	// Network Extensions
+	// ---- 
+	/// Customized or optimized  
+	NetworkCustom(NetworkId, PolicyId),
+}
+
+pub trait MintableAsset {
+	/// Returns the minting policy of the given asset.
+	///
+	/// If `asset` is knonwn, it will return `None`. 
+	fn minting_policy(asset: T::AssetId) -> Option<MintingPolicy>;
+}
+```
+
+Please keep in mind `AccountId` can be **a multi-signature account with threshold N of M** (if
+the underlying network supports it), so minting operation could be delegated on a secure group of
+accounts. That case covers use cases like bridges of wrapped elements, like _Merchants_ in _WBTC_.
+
+If issuer is a **smart contract**, the minting could be more flexible like _MakerDAO_ and its _DAI_
+token.
+
+The policy `OneTimeIssuer(..)` ensures that the token has no inflation because the
+administrator/operator cannot mint more tokens later. After the first minting operation, that policy
+becomes [_immutable_](#mutable-policy).
+
+The **Network extensions** is a mechanism allows network customizations or optimizations on
+specialized parachains. Some parachains could create complex native policies that will be cheaper
+than smart-contract ones.
+
+### Transfer Policy
+
+**NOTE @miguel:** Review ERC-1400 and security token transfer restrictions. It will need a customized
+field too.
+
+
+### Mutability Policy
+
+The mutability of a policy indicates if other policies could change in the future. Once a policy has
+been set as immutable, no one can revert it.  The aim of this policy is the improvement of
+transparency on the asset. For instance, an immutable `NotAllowed` minting policy ensures that the
+token won't support any inflation.
+
+```Rust
+	struct Mutability {
```

Comment:
```suggestion
	enum Policy {
```

Otherwise it is very confusing

and maybe you will also want something like 

```
enum Mutability {
    Immutable,
    Mutable { by: AccountId },
}
```

Reply:
Scoped mutability is an interesting idea

Reply:
Having the attribute of mutability for a policy is confusing to me. Stuff will happen and policies must be updated.
Could something like below considered (having a set of authroties)?
```rust
/// Authorities with privilege to update policy.
/// `None` means no one can update the policy making it immutable.
pub type PolicyMakers = Option<Vec<AccountId>>;
```